src/crush/mapper.c

```diff
@@ -792,7 +792,7 @@ static void crush_choose_indep(const struct crush_map *map,
 							out2, rep,
 							recurse_tries, 0,
 							0, NULL, r, choose_args);
-						if (out2[rep] == CRUSH_ITEM_NONE) {
+						if (out2 && out2[rep] == CRUSH_ITEM_NONE) {
```

Comment:
shall we check for null at line 801, see https://github.com/ceph/ceph/pull/29316/commits/f575e060bcf2ae2ee414035c2435f54b5915ce6a#diff-0df13ad294f6585c322588cfe026d701R801